to.etc.domui/src/main/java/to/etc/domui/component/layout/ErrorMessageDiv.java

```diff
@@ -126,6 +128,14 @@ protected NodeContainer createErrorLine(UIMessage m) {
 		return d;
 	}
 
+	/**
+	 * Adds css selector for additional styling to div container.
+	 */
+	private void addAdditionalStyling(@Nonnull UIMessage m) {
+		if(getChildCount() >= 0 && getVisibility() == VISIBLE)
+			addCssClass("ui-emd-brd-" + m.getType().name().toLowerCase());
```

Comment:
Ok, so the plan is to have the whole component expose a class that represents the "error" inside it?

This code would not work if there are multiple messages with different severity, i.e. one error, one info, one warning. In this case the "last one added" wins and that severity is exposed in the class.

The proper thing to do would be to first add the message, then walk over all messages to see what the highest severity is. Then you set the class to that severity. You also need to make sure that "earlier" classes are removed. The easiest way to do that is to actually have the component have a "severity" too and set that; then you can use the "previous severity" to remove the class and the new one to add it.